src/components/forms/AddressInput/index.tsx

```diff
@@ -38,51 +50,88 @@ const AddressInput = ({
   inputAdornment,
   validators = [],
   defaultValue,
+  value = '',
   disabled,
-}: AddressInputProps): React.ReactElement => (
-  <>
-    <Field
-      className={className}
-      component={TextField as any}
-      defaultValue={defaultValue}
-      disabled={disabled}
-      inputAdornment={inputAdornment}
-      name={name}
-      placeholder={placeholder}
-      testId={testId}
-      text={text}
-      type="text"
-      spellCheck={false}
-      validate={composeValidators(required, mustBeEthereumAddress, ...validators)}
-    />
-    <OnChange name={name}>
-      {async (value: string) => {
-        const address = trimSpaces(value)
-        // A crypto domain name
-        if (isValidEnsName(address) || isValidCryptoDomainName(address)) {
-          try {
-            const resolverAddr = await getAddressFromDomain(address)
-            const formattedAddress = checksumAddress(resolverAddr)
-            fieldMutator(formattedAddress)
-          } catch (err) {
-            logError(Errors._101, err.message)
-          }
-        } else {
-          // A regular address hash
-          let checkedAddress = address
-          // Automatically checksum valid (either already checksummed, or lowercase addresses)
-          if (isValidAddress(address)) {
+}: AddressInputProps): React.ReactElement => {
+  const showNetworkPrefix = useSelector(showShortNameSelector)
+  const [prefixedAddress, setPrefixedAddress] = React.useState(() => addNetworkPrefix(value, showNetworkPrefix))
+  const [prefixedError, setPrefixedError] = React.useState(false)
+
+  return (
+    <>
+      <Field
+        className={className}
+        component={TextField as any}
+        defaultValue={defaultValue}
+        disabled={disabled}
+        inputAdornment={inputAdornment}
+        name={name}
+        inputProps={{
+          value: showNetworkPrefix ? prefixedAddress : value,
```

Comment:
Could you not use an effect, with the user preference as a dependency to adjust the value on render/the preference changes? I'm not sure if it's the best idea to have two sources of truth. This would also remove the requirement of passing the `value` prop.